src/platform/intel/cavs/include/cavs/lib/pm_memory.h

```diff
@@ -154,4 +154,26 @@ void cavs_pm_memory_lp_sram_power_gate(void *ptr, uint32_t size, bool enabled);
 
 #endif /* CONFIG_LP_SRAM */
 
+#if CONFIG_L1_DRAM
+
+static inline void cavs_pm_memory_l1_dram_banks_power_gate(
+	uint32_t start_bank_id, uint32_t ending_bank_id, bool enable)
+{
+	uint32_t mask = MASK(start_bank_id, ending_bank_id);
+	uint32_t pgctl = io_reg_read(L1_MEM_DRAM_PGCTL);
+	uint32_t expected = enable ? 0 : mask;
+
+	if (enable)
+		pgctl &= ~mask;
+	else
+		pgctl |= mask;
+
+	io_reg_write(L1_MEM_DRAM_PGCTL, pgctl);
+
+	while ((io_reg_read(L1_MEM_DRAM_PGISTS) & mask) != expected)
+		idelay(MEMORY_POWER_DOWN_DELAY);
```

Comment:
Will need the timeout here.

Reply:
@lgirdwood you reference old code, the timeout is already added :)

Reply:
Do you still have to push, I'm just seeing
```c
while ((io_reg_read(L1_MEM_DRAM_PGISTS) & mask) != expected)
		idelay(MEMORY_POWER_DOWN_DELAY);
```
So if power status never changes we will just keep spinning.

Reply:
Ohh I know - since the timeout was also not present in the other power gating functions like for HPSRAM & LPSRAM I have added it as a separate commit. So the initial one still keeps the loop without the timeout as it was added in subsequent commit. Anyway I will update the very first one.